packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1144,8 +1155,13 @@ abstract class FlutterCommand extends Command<void> {
         ? stringsArg(FlutterOptions.kDartDefinesOption)
         : <String>[];
 
-    if (argParser.options.containsKey('web-renderer')) {
-      dartDefines = updateDartDefines(dartDefines, stringArgDeprecated('web-renderer')!);
+    WebRendererMode webRenderer = WebRendererMode.autoDetect;
+    if (argParser.options.containsKey(FlutterOptions.kWebRendererFlag)) {
+      final WebRendererMode? mappedMode = _webRendererModeMap[stringArgDeprecated(FlutterOptions.kWebRendererFlag)!];
+      if (mappedMode != null) {
+        webRenderer = mappedMode;
+      }
+      dartDefines = updateDartDefines(dartDefines, webRenderer);
```

Comment:
This happens to work because the condition above is always true. Shouldn't we always update dart defines, even if `--web-renderer` is omitted?